server/src/course/course.recalgo.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Applies stemming rules to reduce a word to its base form
+ */
+const stemWord = (word) => {
+  if (word.endsWith("sses")) {
+    return word.replace(/sses$/, 'ss');
+  } if (word.endsWith("ies")) {
+    return word.replace(/ies$/, 'y');
+  } if (word.endsWith("es") && !/[aeiou]es$/.test(word)) {
+    return word.replace(/es$/, '');
+  } if (word.endsWith("s") && word.length > 1 && !/[sxz]$/.test(word)) {
+    return word.replace(/s$/, '');
+  }
+  return word;
+}
+
+/**
+ * Preprocesses the description to remove pluralities and unnecessary punctuation
+ * @param description A course description that needs to be preprocessed
+ * @returns The processed description for a course
+ */
+export const preprocess = (description: string) => {
+  let sentences = description.match(/[^.!?]*[.!?]\s+[A-Z]/g) || [description];
+  let processedText = sentences.map(sentence => {
+    let words = sentence.match(/\b\w+\b/g) || [];
```

Comment:
Another thought I had about preprocessing was getting rid of "filler words," (i.e. and, the, to, for, with...)

Reply:
nice idea! Also i saw "this" and maybe any pronouns?